Cython/Compiler/ModuleNode.py

```diff
@@ -1692,9 +1692,9 @@ def generate_usr_dealloc_call(self, scope, code):
         code.putln("{")
         code.putln("PyObject *etype, *eval, *etb;")
         code.putln("PyErr_Fetch(&etype, &eval, &etb);")
-        code.putln("++Py_REFCNT(o);")
+        code.putln("__Pyx_SET_REFCNT(o, Py_REFCNT(o) + 1);")
```

Comment:
```suggestion
        # increase the refcount while we are calling into user code to prevent recursive deallocation
        code.putln("__Pyx_SET_REFCNT(o, Py_REFCNT(o) + 1);")
```